components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/AbstractUserInfoResponseBuilder.java

```diff
@@ -291,11 +291,14 @@ private RequestObject getRequestObject(OAuth2TokenValidationResponseDTO tokenRes
         AuthorizationGrantCacheKey cacheKey = new AuthorizationGrantCacheKey(getAccessToken(tokenResponse));
         AuthorizationGrantCacheEntry cacheEntry = AuthorizationGrantCache.getInstance()
                 .getValueFromCacheByToken(cacheKey);
+        RequestObject requestObject = null;
 
         if (cacheEntry != null) {
             cacheEntry.getRequestObject();
+            return cacheEntry.getRequestObject();
+        } else {
```

Comment:
Can assign requestObject = cacheEntry.getRequestObject() inside if, and return directly without having an else statement.